src/Query/QueryBuilder.php

```diff
@@ -1266,7 +1323,14 @@ private function getSQLForSelect(): string
             throw new QueryException('No SELECT expressions given. Please use select() or addSelect().');
         }
 
-        return $this->connection->getDatabasePlatform()
+        $withSQL = match (true) {
+            count($this->withParts) > 0 => $this->connection->getDatabasePlatform()
+                ->createWithSQLBuilder()
+                ->buildSQL(new WithQuery($this->withParts)),
+            default => '',
+        };
+
+        $selectSQL = $this->connection->getDatabasePlatform()
```

Comment:
This `match(true)` statement looks really odd. I think, a good old `if` is fine here.

```suggestion
        $selectSQL = '';
        if (count($this->withParts) > 0) {
            $selectSQL .= $this->connection->getDatabasePlatform()
                ->createWithSQLBuilder()
                ->buildSQL(new WithQuery($this->withParts));
        }

        $selectSQL .= $this->connection->getDatabasePlatform()
```

Reply:
Yep, I agree